poncho/src/poncho_package_run

```diff
@@ -152,6 +152,12 @@ function cleanup {
     s=$((total_ms/1000))
 
     logmsg total runtime: $(printf "%d.%0.3d seconds" "${s}" "${ms}")
+    cpids=$(pgrep -P "$$")
```

Comment:
I don't think pgrep is a standard utility.  Can we get the same information from `ps`?

Reply:
Nevermind, in linux `ps` and `pgrep` come from the same package (procps).

Reply:
I hesitate to ask.... but what if those children forked other processes?

Reply:
Just checked, pgrep is not recursive, bummer.

Reply:
Let's not make the perfect the enemy of the good here, but at least drop a comment indicating that the solution isn't complete..

Reply:
How responsible should a process be for cleaning up its children/grandchildren/great-grandchildren/etc? For example, if we have
work_queue_factory -> poncho_package_run -> work_queue_worker -> cctools_monitor
Should processes be responsible for all their progeny? If so the factory would send termination signals to poncho, the worker, and the resource monitor? Poncho will then also send termination signals to the worker and the monitor, and then the worker will also send a termination signal to the monitor.  It seems a bit wasteful to me.
Or can we rely on each process cleaning up its immediate child. For example the factory sends SIGTERM to poncho, which sends SIGTERM to the worker, which sends SIGTERM to the monitor, monitor returns, worker returns, poncho returns, and finally the factory returns.
If we are looking for a recursive solution like the first one listed here, pstree along with some parsing might be of great assistance.
<img width="726" alt="Screen Shot 2022-02-17 at 9 22 34 PM" src="https://user-images.githubusercontent.com/31227025/154607346-c9b82f41-54e6-4069-955d-9cab19580d14.png">